lib/matplotlib/pyplot.py

```diff
@@ -999,9 +999,24 @@ def gcf() -> Figure:
         return figure()
 
 
-def fignum_exists(num: int) -> bool:
-    """Return whether the figure with the given id exists."""
-    return _pylab_helpers.Gcf.has_fignum(num) or num in get_figlabels()
+def fignum_exists(num: int | str) -> bool:
+    """Return whether the figure with the given id exists.
+
+    Parameters
+    ----------
+    num : int or str
+        A figure identifier.
+
+    Returns
+    -------
+    bool
+        Whether or not a figure with id `num` exists.
```

Comment:
```suggestion
        Whether or not a figure with id ``num`` exists.
```
I _think_ this is the reason for the doc build failure: with single quotes, Sphinx tries to create a link, with double quotes we get monospace formatting.

Reply:
Yes, MPL has defined single quote as linking.

https://matplotlib.org/stable/devel/document.html#id1